operator/api/v1alpha2/keptnevaluationprovider_types.go

```diff
@@ -61,3 +62,7 @@ type KeptnEvaluationProviderList struct {
 func init() {
 	SchemeBuilder.Register(&KeptnEvaluationProvider{}, &KeptnEvaluationProviderList{})
 }
+
+func (p *KeptnEvaluationProvider) HasSecretKey() bool {
+	return p.Spec.SecretKeyRef == (corev1.SecretKeySelector{})
```

Comment:
in this case if the secretKeyRef is set with a non-empty structure, the method will return false, which is not expected behaviour

Reply:
please add a simple unit test when possible